### PR TITLE
CI: Update Cygwin NumPy to 1.26 in Python 3.9

### PR DESCRIPTION
**Pull request by DWesl**
Removes the NumPy-pinning requirement noted in #7403.

Changes proposed in this pull request:

 * NumPy can use the Netlib BLAS interface, and tests against it regularly.  Cygwin provides Netlib interfaces for BLAS and Lapack in `liblapack-devel`.  NumPy recently changed how to specify BLAS versions: [reference here](https://numpy.org/doc/1.26/user/building.html) and [elaboration here](https://github.com/numpy/numpy/blob/main/meson_options.txt).  [Pip can pass arguments through to sub-processes](https://github.com/numpy/numpy/issues/24703#issuecomment-1722379388).  Combining those, it should be possible to allow CI to use the most recent NumPy version.  This is a slight tweak of the fix proposed in #7403.

### CODE REVIEW

**Comment by radarhere** [comment]
> Fixes https://github.com/python-pillow/Pillow/pull/7403.

Ah, ok, you mean that the PR pinned NumPy on Cygwin, and this removes the pinning. Normally, I would only think of issues as being "fixed", not PRs that passed and were merged.

File: .github/workflows/test-cygwin.yml [hunk 1]
```diff
@@ -116,7 +117,7 @@ jobs:
       - name: Upgrade NumPy
         shell: dash.exe -l "{0}"
         run: |
-          python3 -m pip install -U "numpy<1.26"
+          python3 -m pip install -U "numpy" --config-settings=setup-args="-Dblas=blas" --config-settings=setup-args="-Dlapack=lapack"
```
Comment (radarhere) [inline .github/workflows/test-cygwin.yml:120]:
```suggestion
          python3 -m pip install -U numpy --config-settings=setup-args="-Dblas=blas" --config-settings=setup-args="-Dlapack=lapack"
```

**Comment by radarhere** [comment]
I've updated the PR title, as this only affects Python 3.9 - [NumPy 1.24 is the last version to support Python 3.8](https://endoflife.date/numpy#python-support), so this makes no difference for 3.8.

**Comment by radarhere** [comment]
It turns out there is a problem, both in this PR and in main, and for both Python 3.8 and 3.9 - [NumPy is not detected by the tests](https://github.com/python-pillow/Pillow/actions/runs/8248697493/job/22559567810#step:13:4362).

**Comment by DWesl** [comment]
Looks like the switch to the egor-tensin Cygwin install action didn't catch all the places the default install path got used: in particular, the egor-tensin/cleanup-path specification of the new path, including the location of the BLAS and Lapack runtimes.

File: .github/workflows/test-cygwin.yml [hunk 2]
```diff
@@ -81,56 +82,58 @@ jobs:
...
-          bash.exe .ci/install.sh
+          C:/tools/cygwin/bin/bash.exe .ci/install.sh
 
       - name: Upgrade NumPy
-        shell: dash.exe -l "{0}"
+        shell: C:/tools/cygwin/bin/dash.exe -l "{0}"
         run: |
-          python3 -m pip install -U "numpy<1.26"
+          python3 -m pip install -U numpy --config-settings=setup-args="-Dblas=blas" --config-settings=setup-args="-Dlapack=lapack"
+          python3 -c "import numpy as np; print(np.__version__)"
```
Comment (hugovk) [inline .github/workflows/test-cygwin.yml:122]: Shall we remove this debugging line?
```suggestion
```

**Review by hugovk — COMMENTED** [review]
Nice work :)

File: .github/workflows/test-cygwin.yml (hunk 2 above)
Comment (DWesl) [reply .github/workflows/test-cygwin.yml]: This has the added benefit of testing whether I properly fixed the pip cache: the last run took around 30 minutes, this one should take around 15.

**Comment by radarhere** [comment]
I've created https://github.com/python-pillow/Pillow/pull/7880 as an alternative to this.

**Comment by hugovk** [comment]
We've merged https://github.com/python-pillow/Pillow/pull/7880, but thanks for this PR that prompted that one :)

**Pull request closed without merging**